pkg/registry/apis/secret/contracts/inline.go

```diff
@@ -0,0 +1,23 @@
+package contracts
+
+import (
+	"context"
+
+	common "github.com/grafana/grafana/pkg/apimachinery/apis/common/v0alpha1"
+)
+
+// SecureValue storage support
+//
+//go:generate mockery --name InlineSecureValueStore --structname MockInlineSecureValueStore --inpackage --filename inline_mock.go --with-expecter
+
+type InlineSecureValueStore interface {
+	// Check that the request user can reference a secret in the context of a given resource (owner)
+	CanReference(ctx context.Context, owner common.ResourceReference, names ...string) (bool, error)
+
+	// Create inline secure value
+	CreateSecureValue(ctx context.Context, owner common.ResourceReference, value common.RawSecretValue) (string, error)
```

Comment:
🤔  we may want to send the old name with the request and skip creating a *new* resource if the resource is always sending a raw value

That would happen for things like:
https://github.com/grafana/plugin-provisioning/blob/main/provisioning/datasources/aws-iot-twinmaker.yaml#L20 😬 😬 😬